services/provider/server/server.go

```diff
@@ -870,17 +879,29 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 		return nil, status.Errorf(codes.Internal, "Failed to construct status response: %v", err)
 	}
 
-	desiredClientConfigHash := getDesiredClientConfigHash(channelName, storageConsumer)
+	storageCluster, err := s.getStorageCluster(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	inTransitEncryptionEnabled := false
+	if storageCluster.Spec.Network != nil && storageCluster.Spec.Network.Connections != nil &&
+		storageCluster.Spec.Network.Connections.Encryption != nil && storageCluster.Spec.Network.Connections.Encryption.Enabled {
+		inTransitEncryptionEnabled = true
+	}
+
+	desiredClientConfigHash := getDesiredClientConfigHash(channelName, inTransitEncryptionEnabled, storageConsumer)
 
 	return &pb.ReportStatusResponse{
 		DesiredClientOperatorChannel: channelName,
 		DesiredConfigHash:            desiredClientConfigHash,
 	}, nil
 }
 
-func getDesiredClientConfigHash(channelName string, storageConsumer *ocsv1alpha1.StorageConsumer) string {
+func getDesiredClientConfigHash(channelName string, inTransitEncryptionEnabled bool, storageConsumer *ocsv1alpha1.StorageConsumer) string {
 	var arr = []any{
 		channelName,
+		inTransitEncryptionEnabled,
 		storageConsumer.Spec.StorageQuotaInGiB,
 	}
```

Comment:
Please add the new var to the end of the list
```suggestion
	var arr = []any{
		channelName,
		storageConsumer.Spec.StorageQuotaInGiB,
		inTransitEncryptionEnabled,
	}
```